regionServiceClient/lib/cloudregister/registerutils.py

```diff
@@ -0,0 +1,419 @@
+    """Check if the instance is already registerd"""
+    if has_repos(smt_server_name) and __has_credentials(smt_server_name):
+        return 1
+
+    return None
+
+
+# ----------------------------------------------------------------------------
+def clean_hosts_file(domain_name):
+    """Remove the smt server entry from the /etc/hosts file"""
+    new_hosts_content = []
+    content = open('/etc/hosts', 'r').readlines()
+    smt_announce_found = None
+    for entry in content:
+        if '# Added by SMT' in entry:
+            smt_announce_found = True
+            continue
+        if smt_announce_found and domain_name in entry:
+            smt_announce_found = False
+            continue
+        new_hosts_content.append(entry)
+
+    hosts_file = open('/etc/hosts', 'w')
+    for entry in new_hosts_content:
+        hosts_file.write(entry)
+    hosts_file.close()
+
+
+# ----------------------------------------------------------------------------
+def exec_subprocess(cmd):
+    """Execute the given command as a subprocess (blocking)"""
+    try:
+        cmd = subprocess.Popen(
+            cmd,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE
+        )
+        cmd.communicate()
+    except:
+        return 0
+
+    return 1
+
+
+# ----------------------------------------------------------------------------
+def find_equivalent_smt_server(configured_smt, known_smt_servers):
+    """Find an SMT server that is equivalent to the currently configured
+       SMT server, only consider responsive servers"""
+    for smt in known_smt_servers:
+        if smt.get_ip() == configured_smt.get_ip():
+            continue
+        if smt.is_equivalent(configured_smt) and smt.is_responsive():
+            return smt
+
+    return None
+
+
+# ----------------------------------------------------------------------------
+def get_available_smt_servers():
+    """Return a list of available SMT servers"""
+    availabe_smt_servers = []
+    if not os.path.exists(REGISTRATION_DATA_DIR):
+        return availabe_smt_servers
+    smt_data_files = glob.glob(REGISTRATION_DATA_DIR + 'availableSMTInfo*')
+    for smt_data in smt_data_files:
+        availabe_smt_servers.append(get_smt_from_store(smt_data))
+
+    return availabe_smt_servers
+
+
+# ----------------------------------------------------------------------------
+def get_current_smt():
+    """Return the data for the current SMT server"""
+    return get_smt_from_store(__get_registered_smt_file_path())
+
+
+# ----------------------------------------------------------------------------
+def get_smt_from_store(smt_store_file_path):
+    """Create an SMTinstance from the stored data."""
+    if not os.path.exists(smt_store_file_path):
+        return None
+
+    smt_file = open(smt_store_file_path, 'r')
+    u = pickle.Unpickler(smt_file)
+    try:
+        smt = u.load()
+    except:
+        smt_file.close()
+        return None
+
+    smt_file.close()
```

Comment:
Well, not quite like that but I see where you are going and it is reasonable to shorten the implementation.